src/test/scala/com/fulcrumgenomics/umi/VanillaUmiConsensusCallerTest.scala

```diff
@@ -375,6 +375,56 @@ class VanillaUmiConsensusCallerTest extends UnitSpec {
     r2.getReadString shouldBe ("N" * 10)
   }
 
+  it should "generate accurate per-read and per-base tags on the consensus reads" in {
+    val builder = new SamRecordSetBuilder(readLength=10)
+    builder.addFrag("READ1", 0, 1, false).foreach(_.setAttribute(DefaultTag, "AAA"))
+    builder.addFrag("READ2", 0, 1, false).foreach(_.setAttribute(DefaultTag, "AAA"))
+    builder.addFrag("READ3", 0, 1, false).foreach(_.setAttribute(DefaultTag, "AAA"))
+    builder.addFrag("READ4", 0, 1, false).foreach(_.setAttribute(DefaultTag, "AAA"))
+    builder.foreach { rec => rec.setReadString("A" * rec.getReadLength) }
+
+    // Monkey with one of the four reads
+    builder.find(_.getReadName == "READ2").foreach { r =>
+      r.getReadBases()(5) = 'C'.toByte
+      r.getBaseQualities()(7) = 5.toByte
+    }
+
+
+    val consensusCaller = new VanillaUmiConsensusCaller(
+      input = builder.toIterator,
+      header = builder.header,
+      options = cco(minReads = 2, minInputBaseQuality = 20.toByte)
+    )
+
+    consensusCaller.hasNext shouldBe true
+    val consensus = consensusCaller.next()
+    consensusCaller.hasNext shouldBe false
+    consensus.getReadBases.foreach(_ shouldBe 'A'.toByte)
+    consensus.getSignedShortArrayAttribute(ConsensusTags.PerBase.RawReadCount)  shouldBe Array[Short](4,4,4,4,4,4,4,3,4,4)
+    consensus.getSignedShortArrayAttribute(ConsensusTags.PerBase.RawReadErrors) shouldBe Array[Short](0,0,0,0,0,1,0,0,0,0)
+    consensus.getIntegerAttribute(ConsensusTags.PerRead.RawReadCount)     shouldBe 4
+    consensus.getIntegerAttribute(ConsensusTags.PerRead.MinRawReadCount)  shouldBe 3
+    consensus.getFloatAttribute(ConsensusTags.PerRead.RawReadErrorRate) shouldBe (1 / 39.toFloat)
+  }
+
+  it should "not generate per-base tags when turned off" in {
+    val builder = new SamRecordSetBuilder(readLength=10)
+    builder.addFrag("READ1", 0, 1, false).foreach(_.setAttribute(DefaultTag, "AAA"))
+    builder.addFrag("READ2", 0, 1, false).foreach(_.setAttribute(DefaultTag, "AAA"))
+    builder.foreach { rec => rec.setReadString("A" * rec.getReadLength) }
+
+    val caller = new VanillaUmiConsensusCaller(input=builder.toIterator, header=builder.header, options=cco(producePerBaseTags=false))
+
+    caller.hasNext shouldBe true
+    val consensus = caller.next()
+    caller.hasNext shouldBe false
+    consensus.getSignedShortArrayAttribute(ConsensusTags.PerBase.RawReadCount)  shouldBe null
+    consensus.getSignedShortArrayAttribute(ConsensusTags.PerBase.RawReadErrors) shouldBe null
+    consensus.getIntegerAttribute(ConsensusTags.PerRead.RawReadCount)     shouldBe 2
+    consensus.getIntegerAttribute(ConsensusTags.PerRead.MinRawReadCount)  shouldBe 2
+    consensus.getFloatAttribute(ConsensusTags.PerRead.RawReadErrorRate) shouldBe 0.toFloat
```

Comment:
I think about this all the time - I do wonder if we shouldn't have a wrapper for SAMRecord, and an implicit to convert Iterator[SAMRecord] to Iterator[BetterSamRecord] so that the conversion is one-time, not everytime an enhanced method is used.